### DISCUSSION TITLE
VSCode && ExtendScript Debugger: relative path

### DISCUSSION THREAD

**Discussion by nfourteen** [joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script#31] [Q&A]
I've made some modifications to the script and I'm trying to debug the issue. The problem is that VSCode debug dies at `app.open( mockupPSD.file );` when the debugger is executed because it can't find the mockup files. The "mockupPSD.file" shows the relative path as expected and traverses upward as expected from settings config. But I can't get the debugger past opening the mockup file.

When I drop the script in PS, it runs and does open the file, but dies shortly there after. Currently debugging with alerts, but hoping someone here can help with VSCode config. I'm assuming it's a context issue where the debugger is executing versus when it works fine in PS.

I've tried `"cwd": "${workspaceFolder}"`, `"script": "${file}"`, and some other `launch.json` config settings, but those have failed. I'm running a Mac M1 with VSCode emulation in Rosetta and PS 2024.

**Comment by joonaspaakko** [comment]
I don't know if it's all about your changes, but I wonder if it could be related to this #30. What if you try running [version 1.8](https://github.com/joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script/issues/30#issuecomment-1834570293)? It's weird enough that v.1.9 isn't working on Windows specifically, perhaps the debugger lands on the same bug...

If it's unrelated, I might not have anything to offer since when it comes to Adobe script debugging, I only use `alert()`.

**Reply by nfourteen** [reply]
@joonaspaakko I got the debugger running early this morning by using absolute paths in the settings config file. There is still however something going on with the debugger config. For instance, I can't debug a `mockupPath` as a folder because the relative path is interpreted as `/tmp00000001` in the debugging context. 🤷 Good to know `alert()` is used for debugging...I'll just stick with that for relative path issues until I can figure out the debugger.

I did find an issue related to #30 during debugging. Working on that now and will post what I've found on the issue discussion.
